config/src/main/java/org/springframework/security/config/annotation/web/configurers/RememberMeConfigurer.java

```diff
@@ -267,6 +267,19 @@ public void init(H http) throws Exception {
 		validateInput();
 		String key = getKey();
 		RememberMeServices rememberMeServices = getRememberMeServices(http, key);
+		if (rememberMeServices instanceof AbstractRememberMeServices) {
+			 /*
+         	  * To avoid CookieTheftException in PersistentTokenBasedRememberMeServices.processAutoLoginCookie(...),
+         	  * override the key or config it twice like below:
+         	  *
+         	  * <code>
+         	  *     http.rememberMe()
+         	  *      .key(key)
+         	  *      .rememberMeServices(new PersistentTokenBasedRememberMeServices(key, userDetailsService, tokenRepository));
+         	  * </code>
+         	  */
+			key = ((AbstractRememberMeServices) rememberMeServices).getKey();
```

Comment:
We do not want to override the key if the user has specified it in `.key("...")`. We should check if the key is null and only then use the one from `AbstractRememberMeServices`.

Reply:
Done